profiling/src/lib.rs

```diff
@@ -0,0 +1,719 @@
+                module_startup_func: Some(minit),
+                module_shutdown_func: Some(mshutdown),
+                request_startup_func: Some(rinit),
+                request_shutdown_func: Some(rshutdown),
+                info_func: Some(minfo),
+                version: VERSION.as_ptr(),
+                globals_size: std::mem::size_of::<DatadogPhpProfilingGlobals>() as u64,
+                globals_ctor: Some(ginit),
+                globals_dtor: Some(gshutdown),
+                post_deactivate_func: Some(prshutdown),
+                deps: DEPS.as_ptr(),
+                ..Default::default()
+            };
+            #[cfg(php_zts)]
+            {
+                // todo: zts
+                module.globals_id_ptr = &mut zend::datadog_php_profiling_globals_id as *mut c_void;
+            }
+            #[cfg(not(php_zts))]
+            {
+                module.globals_ptr = zend::datadog_php_profiling_globals_get()
+                    as *mut zend::zend_datadog_php_profiling_globals
+                    as *mut c_void;
+            }
+
+            MODULE.write(module);
+        });
+
+        &mut *MODULE.as_mut_ptr()
+    }
+}
+
+static mut PREV_INTERRUPT_FUNCTION: MaybeUninit<Option<zend::VmInterruptFn>> =
+    MaybeUninit::uninit();
+static mut PREV_EXECUTE_INTERNAL: MaybeUninit<
+    unsafe extern "C" fn(execute_data: *mut zend::zend_execute_data, return_value: *mut zend::zval),
+> = MaybeUninit::uninit();
+
+/* Important note on the PHP lifecycle:
+ * Based on how some SAPIs work and the documentation, one might expect that
+ * MINIT is called once per process, but this is only sort-of true. Some SAPIs
+ * will call MINIT once and then fork for additional processes.
+ * This means you cannot do certain things in MINIT and have them work across
+ * all SAPIs, like spawn threads.
+ *
+ * Additionally, when Apache does a reload it will go through the shutdown
+ * routines and then in the same process do the startup routines, so MINIT can
+ * actually be called more than once per process as well. This means some
+ * mechanisms like std::sync::Once::call_once may not be suitable.
+ * Be careful out there!
+ */
+unsafe extern "C" fn minit(r#type: c_int, module_number: c_int) -> ZendResult {
+    #[cfg(debug_assertions)]
+    trace!("MINIT({}, {})", r#type, module_number);
+
+    // SAPI shouldn't ever change
+    let sapi_module = zend::sapi_module;
+    if !sapi_module.name.is_null() {
+        let mut sapi = SAPI.borrow_mut();
+        let sapi_name = CStr::from_ptr(sapi_module.name);
+        *sapi = Sapi::from_name(sapi_name.to_string_lossy().as_ref())
+    }
+
+    /* Use a hybrid extension hack to load as a module but have the
+     * zend_extension hooks available:
+     * https://www.phpinternalsbook.com/php7/extensions_design/zend_extensions.html#hybrid-extensions
+     */
+    // Transfer the module handle to the zend_extension. Extensions have
+    // longer lifetimes than modules in the engine.
```

Comment:
It's not a problem per se, but you have mixed commenting styles in this PR, the `/* */` and `//`. Curious if there's a logic to this? As a reader it's not clear to me that there is.

Reply:
`/*` for multiline but apparently not consistent ^_^ it should be more consistent now.